signer/src/stacks/wallet.rs

```diff
@@ -60,6 +61,9 @@ pub struct SignerWallet {
     network_kind: NetworkKind,
     /// The multi-sig address associated with the public keys.
     address: StacksAddress,
+    /// The next nonce for the StacksAddress associated with the address of
+    /// the wallet.
+    nonce: Arc<AtomicU64>,
```

Comment:
I don't find where this is shared. I could be missing something ofc. Why do we need to have an `Arc` here?

Reply:
We need it to support cloning the struct. Not sure if we want it to implement `Clone` though. 

Note that this keeps the clones tied together, since once a `nonce` is incremented in one of the clones it's incremented everywhere (I'll add a test for this). While desirable and possibly "correct", I also don't like it since it's probably surprising. Maybe I should remove the atomic altogether? Any opinions here?

Reply:
I could also remove the default `Clone` implementation, or manually implement it to make it more "natural".

Reply:
Yeah if we can get rid of the `Clone` that seems like the best option.